cpp/src/arrow/compute/api_scalar.h

```diff
@@ -259,6 +259,15 @@ Result<Datum> IsValid(const Datum& values, ExecContext* ctx = NULLPTR);
 ARROW_EXPORT
 Result<Datum> IsNull(const Datum& values, ExecContext* ctx = NULLPTR);
 
+// ----------------------------------------------------------------------
+// String functions
+
+struct ARROW_EXPORT ContainsExactOptions : public FunctionOptions {
+  explicit ContainsExactOptions(std::string pattern = "") : pattern(pattern) {}
```

Comment:
I don't think defaulting to the empty string makes sense here.

Reply:
This is needed to be able to construct an instance of it in Cython. I can remove this and switch the Cython instantiation to a pointer.